src/main/scala/zio/prelude/Validation.scala

```diff
@@ -293,6 +293,10 @@ object Validation extends LowPriorityValidationImplicits {
   def fromOption[A](value: Option[A]): Validation[Unit, A] =
     value.fold[Validation[Unit, A]](fail(()))(succeed)
 
+  def fromPredicate[E, A](predicate: Boolean, value: => A, error: => E): Validation[E, A] =
```

Comment:
How about this signature:

```scala
def fromPredicate[E, A](error: E, value: A)(f: A => Boolean): Validation[E, A] = ???
```

Or possibly even better:

```scala
def fromPredicate[E, A](value: A)(f: A => Boolean): Validation[None.type, A] = ???
```

Reply:
I think the first one makes sense; allows the user to control the error returned.

Reply:
Maybe we need two versions here, `fromPredicate` that just provides a default failure value and then `fromPredicateWith` that lets the user specify an error to fail with if the predicate is not satisfied.

Reply:
Good idea, I'll add that.